docs/cloud/azure/architecture.md

```diff
@@ -1 +1,939 @@
+local high-performance disk thus increasing performance. The cache
+directory is continuously monitored and a size limit is maintained by
+deleting files according to a LRU (least recently used) algorithm.
+
+Caching coupled with enabling secondary threads can increase the
+performance of queries against a HDB on Azure Blob Storage. The larger
+the number of secondary threads, irrespective of CPU core count, the
+better the performance of kdb+ object storage. Conversely, the
+performance of cached data appears to be better if the secondary-thread
+count matches the CPU core count.
+
+It is recommended to use compression on the HDB data residing on Azure
+Blob Storage. This can reduce the cost of object storage and possible
+egress costs, and also counteract the relatively high-latency and low
+bandwidth associated with Azure Blob Storage.
+
+Furthermore, Azure Blob Storage is useful for archiving, tiering, and
+backup purposes. The TP log file and the sym can be stored each day and
+archived for a period of time. The lifecycle management of the object
+store simplifies clean-up, whereby one can set the expiration time on any
+file. The versioning feature of Azure Blob Storage is particularly
+useful when a sym file bloat happens due to feed misconfiguration or
+upstream change. Migrating back to a previous version saves the health
+of the whole database.
+
+Azure Blob Storage provides strong read-after-write consistency. After a
+successful write or update of an object, any subsequent read request
+immediately receives the latest version of the object. Azure Blob
+Storage also provides strong consistency for list operations, so after a
+write, you can immediately perform a listing of the objects in a bucket
+with all changes reflected. This is especially useful when there are
+many kdb+ processes reading from Azure Blob Storage, as it ensures
+consistency.
+
+A kdb+ feed can subscribe to a Azure Blob Storage file update that the
+upstream drops into a bucket and can start its processing immediately.
+The data is available earlier compared to the solution when the feed is
+started periodically (e.g. in every hour).
+
+### Azure Disk Storage
+
+Azure Disk Storage can be used to store HDB and Tickerplant data, and is
+fully compliant with kdb+. It supports all the POSIX semantics required.
+With the introduction of Azure Ultra Disk volumes, users experience
+increased performance of 300 IOPS/GiB, up to a maximum of 160 K IOPS per
+disk and more durability, reducing the possibility of a storage volume
+failure.
+
+### Azure Files
+
+Azure Files over NFS offers NFS service for nodes in the same
+availability zone, and can run across zones, or can be exposed
+externally. Azure Files can be used to store HDB and tickerplant data
+and is fully compliant with kdb+. NFS support for Azure Files is
+currently in preview. Microsoft are planning on getting it to General
+Availability shortly.
+
+### Lustre FS (placeholder)
```

Comment:
Azure Managed Lustre FS is a valid option now. Details here: https://learn.microsoft.com/en-us/azure/azure-managed-lustre/
We can remove the "placeholder" I think.